pkg/apis/migration/v1alpha1/condition.go

```diff
@@ -158,8 +158,13 @@ func (r *Conditions) DeleteCondition(types ...string) {
 	kept := []Condition{}
 	for i := range r.List {
 		condition := r.List[i]
-		_, found := filter[condition.Type]
-		if !found {
+		_, matched := filter[condition.Type]
```

Comment:
@jortel did you find a bug in the `DeleteCondition` method, or just needed some functionality change for this PR?

Reply:
Found a bug.  The delete should not actually remove the condition while staging.